ocean_lib/web3_internal/web3_overrides/contract.py

```diff
@@ -49,6 +50,9 @@ def transact(self, transaction):
                 raise ValueError(
                     "Please ensure that this contract instance has an address."
                 )
+        if "chainId" not in transact_transaction:
+            chain_id = cf.web3.eth.chain_id
+            transact_transaction["chainId"] = chain_id
```

Comment:
I think you can comprise 54 and 55 into one line, and remove the local assignment.